dms/security/security.xml

```diff
@@ -48,6 +48,19 @@
             name="domain_force"
         >['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
     </record>
+    <record id="rule_file_locked" model="ir.rule">
+        <field name="name">Locked files are only modified by locker user.</field>
+        <field name="model_id" ref="model_dms_file" />
+        <field name="groups" eval="[(4, ref('base.group_user'))]" />
+        <field name="global" eval="True" />
+        <field name="perm_read" eval="0" />
+        <field name="perm_create" eval="1" />
+        <field name="perm_write" eval="1" />
+        <field name="perm_unlink" eval="1" />
+        <field
+            name="domain_force"
+        >['|', ('locked_by', '=', False), ('locked_by', '=', user.id)]</field>
```

Comment:
What about using the newly discovered?

```suggestion
        >[('locked_by', '=?', user.id)]</field>
```

Reply:
After testing, I'll have to remove this one, because it doesn't actually mean the same.

The original domain means "locked_by can be false or user.id", but the new one means "if user.id is False, we don't filter by locked_by; but if it has a value, we have to filter by it".

Reply:
I see. Indeed it's not the same. The operator servers for other purpose.